indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -519,20 +636,40 @@ public SeekableStreamSupervisor(
     this.useExclusiveStartingSequence = useExclusiveStartingSequence;
     this.dataSource = spec.getDataSchema().getDataSource();
     this.ioConfig = spec.getIoConfig();
+    this.dynamicAllocationTasksProperties = ioConfig.getDynamicAllocationTasksProperties();
```

Comment:
I modified the way of obtaining configurations from Map to new interface `AutoScalerConfig` with a default impl `DefaultAutoScaleConfig`. So that we can use JackSon to Instantiate a Config with **default values** instead of map.get/parse everywhere. **Also ensure consistency of default values.**

In this way, we don't need to get a handle to `SupervisorTaskAutoscaler`, just `autoScalerConfig.getTaskCountMax()` :)